xain/grpc/participant.py

```diff
@@ -81,18 +128,27 @@ def run():
 
     # start the heartbeat in a different thread
     terminate_event = threading.Event()
+    selected_event = threading.Event()
     heartbeat_thread = threading.Thread(
-        target=heartbeat, args=(channel, terminate_event)
+        target=heartbeat, args=(channel, terminate_event, selected_event)
     )
     heartbeat_thread.start()
 
-    try:
-        # never returns unless there is an exception
-        heartbeat_thread.join()
-    except KeyboardInterrupt:
-        terminate_event.set()
-        channel.close()
+    # standby:
+    standby(channel, p, terminate_event, selected_event)
+
+    # try:
```

Comment:
I think it's helpful to keep this logic if we want to shutdown the participant without a `FINISHED` event